plugins/package-configuration-providers/ort-config/src/main/kotlin/OrtConfigPackageConfigurationProvider.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2023 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.plugins.packageconfigurationproviders.ortconfig
+
+import java.io.File
+
+import org.apache.logging.log4j.kotlin.Logging
+
+import org.ossreviewtoolkit.downloader.vcs.Git
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.Provenance
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.VcsType
+import org.ossreviewtoolkit.model.config.PackageConfiguration
+import org.ossreviewtoolkit.model.utils.PackageConfigurationProvider
+import org.ossreviewtoolkit.plugins.packageconfigurationproviders.api.DirectoryPackageConfigurationProvider
+import org.ossreviewtoolkit.plugins.packageconfigurationproviders.api.PackageConfigurationProviderFactory
+import org.ossreviewtoolkit.utils.common.safeMkdirs
+import org.ossreviewtoolkit.utils.ort.ortDataDirectory
+
+private const val ORT_CONFIG_REPOSITORY_BRANCH = "main"
+private const val ORT_CONFIG_REPOSITORY_URL = "https://github.com/oss-review-toolkit/ort-config.git"
+private const val PACKAGE_CONFIGURATIONS_DIR = "package-configurations"
+
+class OrtConfigPackageConfigurationProviderFactory : PackageConfigurationProviderFactory<Unit> {
+    override val type = "OrtConfig"
+
+    override fun create(config: Unit): PackageConfigurationProvider = OrtConfigPackageConfigurationProvider()
+
+    override fun parseConfig(config: Map<String, String>) = Unit
+}
+
+/**
+ * A [PackageConfigurationProvider] that provides [PackageConfiguration]s from the
+ * [ort-config repository](https://github.com/oss-review-toolkit/ort-config).
+ */
+open class OrtConfigPackageConfigurationProvider : PackageConfigurationProvider {
```

Comment:
I believe this should be non-`open`. (I'll later provide an alignment PR that does the same for `OrtConfigPackageCurationProvider`.)